android/app/src/main/java/app/organicmaps/MwmActivity.java

```diff
@@ -1983,6 +1952,44 @@ private void onLocationPermissionsResult(java.util.Map<String, Boolean> permissi
       if (requestedForRecording && LocationUtils.checkFineLocationPermission(this))
         startTrackRecording();
 
+      if (LocationUtils.checkFineLocationPermission(this))
```

Comment:
here and in lines above its being checked 2-3 times, please create a final var instead

(and looking at the code snippets in https://developer.android.com/develop/sensors-and-location/location/permissions/runtime#java if FINE is granted then no need to check for COARSE, i.e. 
```
  public static boolean checkLocationPermission(@NonNull Context context)
  {
    return checkFineLocationPermission(context) || checkCoarseLocationPermission(context);
  }
```
is redundant and could be replaced with just the COARSE check